### ISSUE TITLE
[IOS] ScrollView with TextInput causes weird scroll position behavior

### ISSUE THREAD

**Issue by bumjin1013** [software-mansion/react-native-screens#2590]
https://github.com/user-attachments/assets/f218c28a-a1d1-4217-af6a-6e58aec88b16

### Description

After enabling the new architecture, upgrading navigation to v7, and upgrading to react-native-screens v4.4.0, the following issues occur:

When a ScrollView contains a TextInput, the scroll behaves abnormally, and the components are re-rendered incorrectly (flickering).

This issue is only observed on iOS and does not occur on Android. The app was working fine before upgrading the React Native version, as well as the versions of react-native-screens and react-navigation.

* Everything was ok before using EnableNewArchitecture = false

react-native:  v0.76.4 
react-native-screens: v4.4.0
 "@react-navigation/bottom-tabs": "^7.2.0",
    "@react-navigation/material-top-tabs": "^7.1.0",
    "@react-navigation/native": "^7.0.14",
    "@react-navigation/native-stack": "^7.2.0",

### Steps to reproduce

1. use rn 0.76.4, rn-screens v4.4.0, react-navigation v7
2. make scrollView with textinput 

### Snack or a link to a repository

non

### Screens version

4.4.0

### React Native version

0.76.4

### Platforms

iOS

### JavaScript runtime

JSC

### Workflow

React Native (without Expo)

### Architecture

Fabric (New Architecture)

### Build type

Debug mode

### Device

Real device

### Device model

iphone 15 pro

### Acknowledgements

Yes

**Comment by leo0grau** [comment]
I am having this same problem, there is any solution ?

**Comment by bumjin1013** [comment]
@leo0grau you can use react-navigation/stack or disable newarchitecture.

**Comment by leo0grau** [comment]
@khedewyAmj , i solved this problema creating a custom header, it solved for me

**Comment by khedewyAmj** [comment]
@leo0grau  Yes , i solved it with remove header left or right i was put in code with use the header it's works fine , thanks you

**Comment by sniperosx** [comment]
Same problem here. Even after upgrading to the latest version. Has somebody found a workaround? Going back to old architecture cannot be the only solution 🤯

@leo0grau What did you do exactly by introducing a "custom header". In navigation? Flat list?

**Comment by leo0grau** [comment]
@sniperosx From my tests, this flickering occurs when there is a scroll (ScrollView/FlatList) and the default header from React Navigation is displayed on the screen. To solve this, I created a custom header, and the problem disappeared.

**Comment by vitas61** [comment]
I am having this same problem, there is any solution?
"react-native": "0.77.1",
"@react-navigation/native": "^7.0.15",
"@react-navigation/native-stack": "^7.2.1",
"react-native-screens": "4.9.1"

Fixes the problem only @react-navigation/stack or disabled or custom header on @react-navigation/native-stack
